data/transactions/logic/parsing.go

```diff
@@ -0,0 +1,105 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package logic
+
+import (
+	"encoding/base32"
+	"encoding/base64"
+	"encoding/binary"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/algorand/avm-abi/abi"
+	"github.com/algorand/go-algorand/data/basics"
+)
+
+// AppCallBytes represents an encoding and a value of an app call argument.
+type AppCallBytes struct {
+	Encoding string `codec:"encoding"`
+	Value    string `codec:"value"`
+}
+
+// NewAppCallBytes parses an argument of "form encoding:value" to AppCallBytes.
+func NewAppCallBytes(arg string) (AppCallBytes, error) {
```

Comment:
One could get that impression in our codebase, but I have never seen that recommended elsewhere.  I think it breaks down quickly because many types are "reference types" without looking like it.  For example, `errors.New` returns `error`, not `*error`.  Of course, maybe that `error` implementation is a pointer.  Who knows?

Other times, you might return a slice type.  That is, effectively, a pointer in the sense that it can be modified in place. (Same for a map).  So should you use `New` or `Make`?  In short, I don't think it's a good distinction, and use New exclusively.  I am open to codifying any principled rule that we use in our code base, but I don't think a simply "New" for pointer types works.